Aplib.Core.Tests/LiftingTests.cs

```diff
@@ -0,0 +1,176 @@
+    /// When compared to the latter, manually created to be trivially correct,
+    /// Then the two should be equal, but the metadata name should differ.
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(MetadataMemberData))]
+    public void ImplicitLifting_ActionLiftedToTactic_DoesNotDifferFromManualTactic(Metadata? metadata)
+    {
+        // Arrange
+        System.Action<IBeliefSet> effect = _ => { };
+        Action<IBeliefSet> action = metadata is null ? new(effect) : new(metadata, effect);
+
+        // Act
+        Tactic<IBeliefSet> liftedTactic = action;
+        Tactic<IBeliefSet> manualTactic = new PrimitiveTactic<IBeliefSet>(action.Metadata, action, _ => true);
+
+        // Assert
+        liftedTactic.Should().BeEquivalentTo(manualTactic, config => config
+            .Excluding(o => o.Metadata.Id)
+            .Excluding(o => o.Metadata.Name));
+        liftedTactic.Metadata.Id.Should().NotBe(manualTactic.Metadata.Id);
+        liftedTactic.Metadata.Name.Should().NotBe(manualTactic.Metadata.Name);
+    }
+
+    /// <summary>
+    /// Given a goal lifted to a goal structure,
+    /// When compared to the latter, manually created to be trivially correct,
+    /// Then the two should be equal, but the metadata name should differ.
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(MetadataMemberData))]
+    public void ImplicitLifting_GoalLiftedToGoalStructure_DoesNotDifferFromManualTactic(Metadata? metadata)
+    {
+        // Arrange
+        Goal<IBeliefSet> goal = metadata is null
+            ? new(It.IsAny<Tactic<IBeliefSet>>(), _ => true)
+            : new(metadata, It.IsAny<Tactic<IBeliefSet>>(), _ => true);
+
+        // Act
+        GoalStructure<IBeliefSet> liftedGoalStructure = goal;
+        GoalStructure<IBeliefSet> manualGoalStructure = new PrimitiveGoalStructure<IBeliefSet>(goal.Metadata, goal);
+
+        // Assert
+        liftedGoalStructure.Should().BeEquivalentTo(manualGoalStructure, config => config
+            .Excluding(o => o.Metadata.Id)
+            .Excluding(o => o.Metadata.Name));
+        liftedGoalStructure.Metadata.Id.Should().NotBe(manualGoalStructure.Metadata.Id);
+        liftedGoalStructure.Metadata.Name.Should().NotBe(manualGoalStructure.Metadata.Name);
+    }
+
+    /// <summary>
+    /// Given a goal structure lifted to a desire set,
+    /// When compared to the latter, manually created to be trivially correct,
+    /// Then the two should be equal, but the metadata name should differ.
+    /// </summary>
+    [Theory]
+    [MemberData(nameof(MetadataMemberData))]
+    public void ImplicitLifting_GoalStructureLiftedToDesireSet_DoesNotDifferFromManualTactic(Metadata? metadata)
+    {
+        // Arrange
+        GoalStructure<IBeliefSet> goalStructure = metadata is null
+            ? new PrimitiveGoalStructure<IBeliefSet>(It.IsAny<Goal<IBeliefSet>>())
+            : new PrimitiveGoalStructure<IBeliefSet>(metadata, It.IsAny<Goal<IBeliefSet>>());
```

Comment:
Conditionals in a test is a test smell, could you separate them into multiple tests? I suggest looking at [how Luka did it](https://github.com/team-zomsa/aplib.net/blob/dev/Aplib.Core.Tests/CombinatorTests.cs).

Reply:
Test smell, whilst it heavily reduces code duplication and no logic relies on this logic.

Reply:
@joachimdekker! I choose you!!!
![pokemon-i-choose-you](https://github.com/team-zomsa/aplib.net/assets/35835627/0f21d7fb-094e-44fd-a01f-1f9de9f35051)
@joachimdekker, use UNIT TEST KNOWLEDGE!!!!

Reply:
@joachimdekker